warehouse-management-service/pkg/database/postgres/shelf.go

```diff
@@ -0,0 +1,198 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	wms "warehouse-management-service"
+)
+
+type shelfQueries interface {
+	createShelfTx(ctx context.Context, tx *sql.Tx, shelf wms.Shelf) error
+	getShelfByIdTx(ctx context.Context, tx *sql.Tx, id string) (wms.Shelf, error)
+	updateShelfTx(ctx context.Context, tx *sql.Tx, shelf wms.Shelf) error
+	deleteShelfTx(ctx context.Context, tx *sql.Tx, id string) error
+	shelfBlockExistsTx(ctx context.Context, tx *sql.Tx, id string) (bool, error)
+}
+
+type shelfQueriesImpl struct{}
+
+type ShelfService struct {
+	queries shelfQueries
+	db      *sql.DB
+}
+
+func NewShelfService(db *sql.DB) *ShelfService {
+	return &ShelfService{
+		queries: new(shelfQueriesImpl),
+		db:      db,
+	}
+}
+
+var InvalidShelfBlock = errors.New("invalid shelfBlockId")
+
+func (s *ShelfService) GetShelfById(ctx context.Context, id string) (wms.Shelf, error) {
+	tx, err := s.db.BeginTx(ctx, nil)
```

Comment:
Why are individual queries executed as part of a transaction?
If you did you do this for learning transactions, that's ok. Otherwise, there's no need for single queries to be part of a transaction.